src/instruction-search.njk

```diff
@@ -0,0 +1,138 @@
+---
+title: Instruction Search
+permalink: search/index.html
+---
+<h1>Instruction Search</h1>
+
+<p>Search by mnemonic or long name. Hit Enter on an exact mnemonic match to jump straight to the instruction details.</p>
+
+<input id="search" placeholder="Search mnemonic, e.g. addi" style="padding:8px; width: 100%; max-width: 480px;" />
+
+{# Creates box divide between checkboxes and rest of page #}
+<fieldset id="extension-filters-fieldset" class="filters">
+  <legend>Filter by extension</legend>
+  <div id="extension-filters" class="filter-options"></div>
+</fieldset>
+
+<ul id="list">
+{% for inst in instructions %}
+  <li
+    data-search="{{ (inst.name ~ ' ' ~ inst.longName) | lower }}"
+    data-mnemonic="{{ inst.name | lower }}"
+    data-extension="{{ inst.extension }}"
+    data-extension-slug="{{ inst.extensionSlug }}"
+  >
+    <a href="/instructions/{{ inst.name }}/">{{ inst.name }}</a>
+    <span class="ext-name" data-extension-slug="{{ inst.extensionSlug }}" title="Extension">
+    - {{ inst.extension }}
+    </span>
+  </li>
+{% endfor %}
+</ul>
+
+<script>
+const input = document.getElementById('search');
+const list = document.getElementById('list');
+const filtersFieldset = document.getElementById('extension-filters-fieldset');
+{# the blank area where we’ll insert checkboxes #}
+const filtersContainer = document.getElementById('extension-filters');
+const items = Array.from(list.querySelectorAll('li'));
+const activeExtensions = new Set();
+
+const extensionSet = new Set();
```

Comment:
I still don't really understand why we are using a `Set` here. It was previously a `Map`. Can we just construct an array and then sort it?